### PR TITLE
Allennlp Predictor class that mimics gec_model

### PR DESCRIPTION
**Pull request by ksteimel**
Sorry for the large PR. These changes introduce a gec_predictor class that is able to mimic the behavior of gec_model but in a way that follows the allennlp api. 
This does handle multi-iteration prediction. However, gec_model had an option for ensemble prediction and that is currently not handled here. 
I have also created a model archive that can be used with this new predictor. You can see the config file for this in test_fixtures/roberta_model. The weights file is downloaded the appropriate location and shared between the predictor and gec_model.. I can tar the folder up and then add it to a public s3 bucket after this PR is approved.

### CODE REVIEW

File: test_fixtures/roberta_model/vocabulary/d_tags.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CORRECT
```
Comment (Frost45) [inline test_fixtures/roberta_model/vocabulary/d_tags.txt:1]: Are the files in this directory the same as the ones in `data/output_vocabulary`?

Comment (ksteimel) [reply test_fixtures/roberta_model/vocabulary/d_tags.txt:1]: yep they are! Should I make a commit to delete the files from data/output_vocabulary?

Comment (Frost45) [reply test_fixtures/roberta_model/vocabulary/d_tags.txt:1]: Yeah, I think so. We need to do an overhaul of the README anyway, so we can just point to this directory for the `vocab_path`.

Comment (Frost45) [reply test_fixtures/roberta_model/vocabulary/d_tags.txt:1]: Actually, I think Damien and I can do this in the next PR. The vocab path will need to be changed in all the tests.

**Comment by ksteimel** [comment]
Sorry! I forgot to add Nitin and Damien to the list of reviewers somehow.

**Review by desilinguist — APPROVED** [review]
I am not that familiar with this codebase so I am going to approve based on what I see and let @Frost45 and @damien2012eng do a more careful review.

**Comment by damien2012eng** [comment]
Great work, @ksteimel !  
Just wondering, the intension of this PR is not to replace the `gec_model`, but to introduce `gec_predictor` and the config file. Am I understanding it correctly?

**Review by Frost45 — APPROVED** [review]
This is amazing work @ksteimel! Thank you for taking this on! Just a few minor comments.

**Review by damien2012eng — APPROVED** [review]

File: gector/gec_predictor.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+from allennlp.models import Model
+from utils.helpers import START_TOKEN
+
+
+@Predictor.register("gec-predictor")
+class GecPredictor(Predictor):
+    """
+    A Predictor for generating predictions from GECtor.
+
+    Note that currently, this is unable to handle ensemble predictions.
```
[the remark this answers is not in the document]
Comment (ksteimel) [reply gector/gec_predictor.py:22]: yes it is not able to work with that. 
I could update that in a future PR to do ensemble models.

File: gector/gec_predictor.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    """
+    def __init__(self, model: Model,
+                 dataset_reader: DatasetReader,
+                 language: str = 'en_core_web_sm',
+                 iterations: int = 3) -> None:
+        super().__init__(model, dataset_reader)
+        #self._tokenizer = SpacyWordSplitter(language=language, pos_tags=True)
```
[the remark this answers is not in the document]
Comment (ksteimel) [reply gector/gec_predictor.py:29]: Yeah haha I should have removed it before opening the PR 😅

File: tests/test_gec_predictor.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+from allennlp.predictors import Predictor
+from allennlp.modules.text_field_embedders import BasicTextFieldEmbedder
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.data import Token
+from allennlp.data.instance import Instance
+from allennlp.data.fields import TextField
+from allennlp.data.dataset import Batch
+
+from gector.gec_predictor import GecPredictor
+from gector.gec_model import GecBERTModel
```
[the remark this answers is not in the document]
Comment (ksteimel) [reply tests/test_gec_predictor.py:16]: i'll remove the duplicate imports

File: gector/gec_predictor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Frost45) [reply gector/gec_predictor.py:20]: That's alright! It means that for now, we shouldn't remove GecModel, even after this PR is merged.

**Pull request merged**